src/js/dp/search.js

```diff
@@ -8,138 +8,173 @@ const fetchHtml = async (url) => {
-        topFilter.checked = theChildren.some(x => x.checked);
-      })
-  });    
-});
-
-// create listeners for the sort dropdown
-const sortSelector = searchContainer.querySelector(".ons-input--sort-select");
-if(!!sortSelector) {
-  sortSelector.addEventListener("change", async (e) => {
-    let strParams = window.location.search;
-    // remove old param
-    strParams = strParams.replace(new RegExp(`[?&]sort\=[^&]+`), "")
-    // replace
-    strParams += `&sort=${e.target.value}`;
-    switchSearchMarkup(strParams, true)
+        switchCheckbox([
+          { isChecked: e.target.checked, filterName: e.target.value },
+        ]);
+        topFilter.checked = theChildren.some((x) => x.checked);
+      });
+    });
   });
-}
 
-// create listeners for the pagination
-const initPaginationListeners = () => {
-  const paginationItems = searchContainer.querySelectorAll(".ons-pagination__item a[data-target-page]");
-  if(!!paginationItems) {
-    paginationItems.forEach((item) => {
-      item.addEventListener("click", async (e) => {
-        e.preventDefault();
-        let strParams = window.location.search;
-        const { targetPage } = e.target.dataset;
-        if( !targetPage ) return;
-        // remove old param if it's there
-        strParams = strParams.replace(new RegExp(`[?&]page\=[^&]+`), "");
-        // add the new page target
-        strParams += `&page=${targetPage}`;
-        switchSearchMarkup(strParams, false, true)
-      });
+  // create listeners for the sort dropdown
+  const sortSelector = searchContainer.querySelector(".ons-input--sort-select");
+  if (!!sortSelector) {
+    sortSelector.addEventListener("change", async (e) => {
+      let strParams = window.location.search;
+      // remove old param
+      strParams = strParams.replace(new RegExp(`[?&]sort\=[^&]+`), "");
+      // replace
+      strParams += `&sort=${e.target.value}`;
+      switchSearchMarkup(strParams, true);
     });
   }
-}
 
-initPaginationListeners();
-
-// filter menu for mobile
-// if the page is running javascript let's make the filter menus togglable and full-screen when displayed
-const toggleBtns = [...searchContainer.querySelectorAll('.search__filter__mobile-filter-toggle')]
-const filterMenu = searchContainer.querySelector('#search-filter')
-filterMenu.classList.add('js-fullscreen-filter-menu-content', 'hide--sm');
-toggleBtns.map((btn) => {
-  btn.classList.remove('hide');
-  btn.addEventListener('click', () => {
-      if( filterMenu.classList.contains('hide--sm') ) {
-          filterMenu.classList.remove('hide--sm');
+  // create listeners for the pagination
+  const initPaginationListeners = () => {
+    const paginationItems = searchContainer.querySelectorAll(
+      ".ons-pagination__item a[data-target-page]"
+    );
+    if (!!paginationItems) {
+      paginationItems.forEach((item) => {
+        item.addEventListener("click", async (e) => {
```

Comment:
`async` unnecessary because it doesn't `await` anything or need to implicitly return a promise.